common-test/src/main/java/io/customer/commontest/config/DefaultTestConfig.kt

```diff
@@ -0,0 +1,32 @@
+package io.customer.commontest.config
+
+/**
+ * Default implementation of [TestConfig] that contains a list of test arguments
+ * and DI graph configuration.
+ * The configuration can be combined with other configurations using the [plus] operator.
+ */
+class DefaultTestConfig internal constructor(
```

Comment:
`TestConfigBuilder` seems to be doing same thing? maybe we can simplify by just utilizing it and not adding another layer on top of it?

Reply:
something like 

```
class TestConfigBuilder {
    private val arguments = mutableListOf<TestArgument>()
    private val diGraph = DIGraphConfiguration()

    fun argument(arg: TestArgument) {
        arguments.add(arg)
    }

    fun diGraph(block: DIGraphConfiguration.() -> Unit) {
        diGraph.apply(block)
    }

    fun build(): TestConfig = object : TestConfig {
        override val arguments: List<TestArgument> = this@TestConfigBuilder.arguments
        override val diGraph: DIGraphConfiguration = this@TestConfigBuilder.diGraph

        override fun plus(other: TestConfig): TestConfig = TestConfigBuilder().apply {
            arguments.addAll(this@TestConfig.arguments)
            arguments.addAll(other.arguments)
            diGraph.plus(other.diGraph)
        }.build()
    }
}

// DSL function for creating TestConfig
fun testConfig(block: TestConfigBuilder.() -> Unit): TestConfig =
    TestConfigBuilder().apply(block).build()
```

Reply:
This is one nice way to look at this 😄 I think it might be less flexible, but as long as it works, I have no strong preference. We can simply update implementation here later if need, the callers should not be affected.

Reply:
I tried this, and plus operator needs builder to be concrete which doesn't work well with DataPipelineTestConfig. So leaving it as it is for now, we can improve this later after datapipeline updates if needed.